app/index.js

```diff
@@ -15,22 +15,27 @@ var AspnetGenerator = yeoman.generators.Base.extend({
       defaults: false,
       desc: 'Use the Grunt JavaScript task runner instead of Gulp in web projects.'
     });
+
+    this.argument('type', { type: String, required: false, desc: 'the project type to create' });
+    this.argument('applicationName', { type: String, required: false, desc: 'the name of the application' });
   },
 
 
-  init: function() {
+  init: function () {
     this.log(yosay('Welcome to the marvellous ASP.NET 5 generator!'));
     this.templatedata = {};
   },
 
-  askFor: function() {
-    var done = this.async();
+  askFor: function () {
+    if (!this.type) {
```

Comment:
That's not enough, if argument is in different order (`yo aspnet --grunt web MyApp`), or there is a typo (`yo aspnet nansy MyApp`), there is no feedback and generator will just create invalid output. As there are no test to cover this problem, I think that at least the check should be added.